cli/src/ui.rs

```diff
@@ -57,6 +57,11 @@ impl Write for &BuiltinPager {
     }
 
     fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        // Since minus::Pager holds the rx end internally, push_str() doesn't
+        // fail even after the paging thread gets terminated.
+        if self.dynamic_pager_thread.is_finished() {
```

Comment:
praise: I'm feeling silly that I had completely forgotten about `JoinHandle::is_finished`, and hadn't considered this approach in https://github.com/martinvonz/jj/pull/4187. 😆😭